chain/src/view.rs

```diff
@@ -118,7 +119,7 @@ pub trait StateReadExt: StateRead {
     }
 }
 
-impl<T: StateRead + ?Sized> StateReadExt for T {}
+impl<T: StateRead + StateReadProto + ?Sized> StateReadExt for T {}
```

Comment:
I don't think this change should be necessary, the `StateRead + ?Sized` bound should be enough.